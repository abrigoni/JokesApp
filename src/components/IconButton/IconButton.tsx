import React from 'react';
import {Pressable} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import {IconProps} from 'react-native-vector-icons/Icon';


const IconButton = ({onPress, ...props}: IconProps) => {
  return (
    <Pressable onPress={onPress}>
      <Icon {...props} />
    </Pressable>
  );
};

export default IconButton;
