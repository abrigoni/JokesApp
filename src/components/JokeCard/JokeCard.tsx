import React, { FC } from 'react'
import { View, StyleSheet, Text, StyleProp, ViewStyle } from 'react-native';
import { Colors } from '../../utils/colors';

const styles = StyleSheet.create({
  card: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingTop: 80,
    paddingBottom: 30,
    paddingHorizontal: 12,
    backgroundColor: Colors.purple, //todo: prop/randomize
    borderRadius: 8,
  },
  text: {
    color: Colors.white,
    fontSize: 24,
    fontFamily: 'OpenSans-Bold',
    textAlign: 'center',
  },
  author: {
    fontSize: 22,
    fontFamily: 'IndieFlower-Regular',
    color: Colors.white,
  },
});

interface Props {
  content: string;
  style?: StyleProp<ViewStyle>;
};

const JokeCard: FC<Props> = ({content, style}) => {
  return (
    <View style={[styles.card, style]}>
      <Text style={styles.text}>{content}</Text>
      <Text style={styles.author}>@designhumor</Text>
    </View>
  )
}

export default JokeCard