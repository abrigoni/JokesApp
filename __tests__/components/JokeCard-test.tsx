import 'react-native';
import React from 'react';
import JokeCard from '../../src/components/JokeCard';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import IconButton from '../../src/components/IconButton';

jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon');

describe('Renders', () => {
  it('Renders correctly', () => {
    const json = renderer
      .create(<JokeCard content={'test'} backgroundColor={Colors.orange} />)
      .toJSON();
    expect(json).toMatchSnapshot();
  });
  it('Renders with IconButton', () => {
    const json = renderer
      .create(
        <JokeCard
          content={'test'}
          backgroundColor={Colors.orange}
          BottomIcon={<IconButton iconProps={{name: 'trash', size: 20}} />}
        />,
      )
      .toJSON();
    expect(json).toMatchSnapshot();
  });
});
