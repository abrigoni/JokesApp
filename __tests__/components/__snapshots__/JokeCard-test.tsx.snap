// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 8,
      "flex": 1,
      "justifyContent": "space-between",
      "marginHorizontal": 20,
      "paddingBottom": 30,
      "paddingHorizontal": 20,
      "paddingTop": 80,
      "width": 600,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ecf0f1",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 24,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#ecf0f1",
          "fontFamily": "IndieFlower",
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    @designhumor
  </Text>
</View>
`;

exports[`Renders Renders with IconButton 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 8,
      "flex": 1,
      "justifyContent": "space-between",
      "marginHorizontal": 20,
      "paddingBottom": 30,
      "paddingHorizontal": 20,
      "paddingTop": 80,
      "width": 600,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ecf0f1",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 24,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ecf0f1",
            "fontFamily": "IndieFlower",
            "fontSize": 22,
          },
          undefined,
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      @designhumor
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Icon
        name="trash"
        size={20}
      />
    </View>
  </View>
</View>
`;
